teuthology/orchestra/remote.py

```diff
@@ -300,6 +300,23 @@ def sudo_write_file(self, path, data, **kwargs):
         """
         self.write_file(path, data, sudo=True, **kwargs)
 
+    def is_mounted(self, path):
+        """
+        Check if the given path is mounted on the remote machine.
+
+        This method checks the contents of "/proc/self/mounts" instead of
+        using "mount" or "findmnt" command since these commands hang when a
+        CephFS client is blocked and its mount point on the remote machine
+        is left unhandled/unmounted.
+
+        :param path: path on remote host
+
+        :raises: :class:`RuntimeError`: unexpected error occurred
+        """
+        if not os.path.isdir(path):
+            raise RuntimeError(f'path {path} is not a directory')
+        return path in self.read_file('/proc/self/mounts', stdout=StringIO())
```

Comment:
`/proc/self/mounts` format is well-defined. I would prefer you actually check the second field of each line rather than grep the whole file.

Reply:
That would either require me to run `cat <path> | awk '{print $2}` myself. Isn't this way simpler and therefore better? 


What about `        return f'{ path}' in self.read_file('/proc/self/mounts', stdout=StringIO())` instead?

Reply:
What about `        return f' { path} ' in self.read_file('/proc/self/mounts', stdout=StringIO())` instead?

This would avoid first and last field but not rest.

Reply:
Using regex `"[A-Za-z0-9]* /proc/self/mounts .*"` can also get this done...

Reply:
You can run the regex in python on the string `read_file` returns?

Reply:
I've been staring at this comment too for quite some time, I don't get this one too. Aren't regex run on strings in as well as beyond Python?

Reply:
You are restricting yourself to resolving this by running a single shell command on the remote. You don't need to do that! Just read the file using `self.read_file` and then use Python's regex library to parse it.